app/javascript/packages/document-capture/hooks/use-asset.js

```diff
@@ -0,0 +1,21 @@
+import { useContext } from 'react';
+import AssetContext from '../context/asset';
+
+function useAsset() {
+  const assets = useContext(AssetContext);
+
+  /**
+   * Returns the mapped URL path associated with the given asset path. If the mapped URL path is not
+   * known, the original path is returned.
```

Comment:
What are the use cases we want this original path behavior? Since the only use cases now seem to be URLs we explicitly list, I think I would expect this to throw an error if we don't have a mapping for the URL?

Is it just in tests where we don't want to import and set up the whole whole thing? In those cases, could we just use a special context that just returns a value for everything? Like a hash with a default value?

Reply:
> What are the use cases we want this original path behavior? [...]

I don't know that I had a strong rationale here aside from having some sensible fallback. At the time, I may have imagined this sort of asset mapping generalized as a sort of middleware-esque opt-in override to "default paths", though in how it's used here, it's pretty obvious that the default, if used, would result in totally broken behavior.

As you mention, it could probably be better to change it to throw an error, or to return some empty-ish value (`null`, for example).

The impact on tests may have played a factor here, since I likely followed an approach similar to what was done with [`useI18n`](https://github.com/18F/identity-idp/blob/master/app/javascript/packages/document-capture/hooks/use-i18n.js), to [fall back to the original key](https://github.com/18F/identity-idp/blob/f67bbaa53998bf05c9c0d59a9cde4331fd77ce5e/app/javascript/packages/document-capture/hooks/use-i18n.js#L59), and be able to [reference those text keys in tests](https://github.com/18F/identity-idp/blob/master/spec/javascripts/packages/document-capture/components/file-input-spec.jsx#L205).

>Like a hash with a default value?

It's not a default behavior of objects, but we might be able to use [`Proxy`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) for this.

Reply:
Using `Proxy` only in tests seems fine to me, since we're not able to run automated IE tests right now anyways? Null value works, too, I'd just love to minimize the chances we accidentally ship something and forget to list a key

Reply:
> I'd just love to minimize the chances we accidentally ship something and forget to list a key

We've also got `make check_asset_strings` for this, added in #3936. Though as far as I can tell, it's not yet integrated into the CI process (it currently errors when run 😬 ).

Reply:
Yikes -- I'll take a look at picking that up and running it as a lint, but that seems like a good alternative to error throwing for this

Reply:
https://github.com/18F/identity-idp/pull/4114